components/RatingStar.js

```diff
@@ -0,0 +1,41 @@
+import styled from "styled-components";
+import { useState } from "react";
+
+const StarRating = ({ rating, onRate }) => {
+  const [hoverRating, setHoverRating] = useState(0);
+
+  return (
+    <Stars>
+      {[1, 2, 3, 4, 5].map((star) => (
+        <Star
+          key={star}
+          filled={star <= (hoverRating || rating)}
+          onClick={() => onRate(star)}
```

Comment:
Your `updatePairingRating` need the id and the amount of stars

```suggestion
          onClick={() => onRate(id, star)}
```